### PR TITLE
Print on rank 0 only and use MPI_Abort instead of stop

### PR DESCRIPTION
**Pull request by matt-frey**
This PR
* replaces `print` statements with `mpi_print` such that only MPI rank 0 writes to the terminal.
* adds a new subroutine `mpi_stop` to exit the program without error.
* replaces occurrences of `MPI_Abort` with `mpi_exit_on_error`.

### CODE REVIEW

File: src/3d/fields/field_netcdf.f90 [hunk 1]
```diff
@@ -430,8 +431,7 @@ subroutine read_netcdf_fields(fname, step)
             if (st == -1) then
                 st = n_steps
             else if ((st == 0) .or. (st > n_steps)) then
-                print *, "Step number out of bounds."
-                error stop
+                call mpi_stop("Step number in NetCDF field file is out of bounds.")
```
Comment (sjboeing) [inline src/3d/fields/field_netcdf.f90:434]: Here, mpi_stop has a text argument. Is this different from mpi_print+mpi_stop above?

File: src/3d/stepper/rk4_utils.f90 [hunk 2]
```diff
@@ -204,7 +205,7 @@ function get_time_step(t) result(dt)
 
             if (dt <= zero) then
                 print "(a10, f0.2, a6)", "Time step ", dt, " <= 0!"
```
Comment (sjboeing) [inline src/3d/stepper/rk4_utils.f90:207]: Should this be an mpi_print?

**Review by sjboeing — COMMENTED** [review]
Thanks Matthias. Happy for you to merge this if you have done a test compilation+run.

File: src/3d/fields/field_netcdf.f90 (hunk 1 above)
Comment (matt-frey) [reply src/3d/fields/field_netcdf.f90:434]: I tried to replace all occurrences of `mpi_print` followed by `mpi_stop` with an `mpi_stop` only that takes a text argument.

File: src/3d/stepper/rk4_utils.f90 (hunk 2 above)
Comment (matt-frey) [reply src/3d/stepper/rk4_utils.f90:207]: That is the issue I explained in Slack. I cannot mimic the print signature at the moment. That is why I have to do it here like this.

**Comment by matt-frey** [comment]
Yes, I also explicitly checked if I can still run the 2D model properly.

**Pull request merged**